client/jetpack-connect/controller.js

```diff
@@ -78,6 +78,26 @@ const analyticsPageTitleByType = {
 	antispam: 'Jetpack Anti-spam',
 };
 
+export function partnerCouponRedirects( context, next ) {
+	const queryArgs = new URLSearchParams( context?.query?.redirect );
+	const partnerCoupon = queryArgs.get( 'partnerCoupon' );
+
+	if ( partnerCoupon.startsWith( 'IONOS_' ) ) {
+		const state = context.store.getState();
+		const siteSlug = getSelectedSiteSlug( state );
+		const wpcomUrl =
+			'development' === config( 'env_id' )
+				? 'http://calypso.localhost:3000'
+				: 'https://wordpress.com';
+
+		return navigate(
+			`${ wpcomUrl }/checkout/${ siteSlug }/jetpack_backup_daily?coupon=${ partnerCoupon }`
```

Comment:
Would it make sense to make this outgoing URL controllable by the Jetpack Redirect Manager: PCYsg-pY7-p2 ?
If we ever move the checkout to jetpack.com instead, then this would allow us to control the coupon redirect location without requiring users to update the plugin

There is an example in `projects/plugins/jetpack/_inc/client/recommendations/prompts/product-suggestion/index.jsx:25` (`generateCheckoutLink()`)

Reply:
For some reason I though this was in the plugin; my mad 😂 
We always control the latest version of what people see in calypso, so skip this comment!

Reply:
Yeah, I like this idea. I'm not sure if it'll be that straight-forward in Calypso, but I'll look into it and add if feasible.

Reply:
I don't even think you need to @rcoll. Since we 100% control Calypso, this should be a pretty easy fix if needed in the future. I was 100% in "Jetpack Plugin" mode 😄